inst/scripts/make-TreeSE-object.R

```diff
@@ -0,0 +1,45 @@
+
+library(tidyverse)
+library(remotes)
+library(mia)
+library(BiocManager)
+library(SummarizedExperiment)
+library(biomformat)
+library(TreeSummarizedExperiment)
+
+
+#Sample phenotype data
+samples <- read.csv("git_metadata_and_community_phenotypes.csv")
+
+#Abundance table
+counts <- readRDS("git_ASV_table.RDS")
+
+#Taxonomic mapping table
+tax <- counts %>% select(3:8)
+
+# In order to make sure that the samples match between the abundance and phenodata tables
+counts_trimmed <- counts[, samples$sample]
+
+
+se <- SummarizedExperiment(assays = list(counts = counts_trimmed),
+                           colData = samples,
+                           rowData = tax)
+
+tse <- as(se, "TreeSummarizedExperiment")
+
+#Phylogenetic tree
+tree <- ape::read.tree("philr_tree_139_asv.nwk")
+
+
+tse <- TreeSummarizedExperiment(assays = list(counts = counts_trimmed),
+                           colData = samples,
+                           rowData = tax,
+                           rowTree = tree)
+
+#sequence information
+refSeq <- DNAStringSet(counts$asv_sequence, start=NA, end=NA, width=NA, use.names=TRUE)
+
+referenceSeq(tse) <- refSeq

```

Comment:
Is this possible to include directly in the first `tse` definition (lines 34-38)? If yes, perhaps it could be there already. Otherwise, does not matter really.

Reply:
I actually wanted to write it separately because it doesn't work all the time. A few times I got it correct, but most of the time, it gives the error "length(s) of 'referenceSeq' (10720) must equal nb of rows in 'x' (0)". I am going to ask to miaverse channel tomorrow if i cannot find the solution

Reply:
Well the error clearly stated: The refernce sequence contains 10720 separate sequences and the TSE contains no rows. 

My guess would be that there is bug depending on the R environment you run the script in. Did you try clearing it and starting with a fresh R session?

Reply:
I assume there is mismatching between metadata and tree data, at least that's what Mr.Borman and I think for now.